### ISSUE TITLE
Some additional features for nested sampling-based analyses

### ISSUE THREAD

**Issue by mreboud** [eos/eos#573]
- [x] Add a seed to nested sampling, check whether specifying it is implemented in `dynesty`.
- [ ] Find a way of recording the nested sampling settings in the analysis file.
- [x] The `sample_nested` task output should also be fixed. E.g. the `nested` directory is not used anymore and the log file is usually empty
- [x] We should save the state of the sampling after every Nth steps, probably by replacing the existing `run` loop that dynesty provides by our own loop.
- [x] Make sure that the `log` file contains relevant information on the status of the sampling. It should e.g. be clear whether the tasks finished or not.
- [x] Give the possibility to require a minimal number of samples. 

In the long term:
- [ ] Devise a way of interfacing an EOS/pypmc mixture with external software, e.g. `Wilson`.

**Comment by fnovak42** [comment]
- [ ] Rename the `eos.data.ImportanceSamples` to something like "dynesty_samples," not just "samples"

**Comment by fnovak42** [comment]
The problem that occurs when using `rstate=np.random.mtrand.RandomState(SEED)` as an input in the dynesty sampler is `AttributeError: 'numpy.random.mtrand.RandomState' object has no attribute 'integers'` This is not an issue when using e.g. `rstate = np.random.default_rng(SEED)` and not noticeable when doing the same thing with `dynesty.utils.resample_equal`.

**Comment by mreboud** [comment]
Can you investigate a bit further?

**Comment by fnovak42** [comment]
I think the best way to solve this is to specify `rstate = np.random.Generator(np.random.MT19937(seed))` in https://github.com/eos/eos/blob/982f9b2ec255619b7b5247508a67819eb2df9f9c/python/eos/analysis.py#L618

**Comment by mreboud** [comment]
> I think the best way to solve this is to specify `rstate = np.random.Generator(np.random.MT19937(seed))` in...

Yes it seems that dynesty's rng is called `rstate` but they expect a `Generator` while pypmc rng is called `rng` and expect a random state... Your solution looks fine to me!

Can you create a branch issue573 on eos/eos and implement this first commit?

**Comment by fnovak42** [comment]
Regarding points 3 and 5, the log file can be moved to `dynesty_results` with just a simple message such as "A dlogz of ... was reached which falls below the threshold initially specified" or "the sampling was stopped since the maximum number of iterations was reached" for now. Later this can extended using information from point 4. Do you agree or would you add more information to the log file?

**Comment by dvandyk** [comment]
It would probably be a good idea to switch from `run_nested` to running the loops ourselves, as documented [here](https://dynesty.readthedocs.io/en/stable/quickstart.html#running-externally).

**Comment by mreboud** [comment]
Another approach to point 6 would be to implement a modified stop_function (https://dynesty.readthedocs.io/en/latest/api.html#dynesty.dynamicsampler.stopping_function) which, passed to the sampler, allows to play on the minimal number of samples.
